syntax = "proto3";

package great;

service DbService {
  rpc GetUserById (Id) returns (ResponseUser);
  rpc RegisterUser (NewUser) returns (Status);
  rpc LoginUser (NewUser) returns (ResponseUser);
}

message Status{
  bool is_done = 1;
  string info = 2;
}

message Id{
  optional Status status = 1;
  int32 id = 2;
}

message ResponseUser{
  Status status = 1;
  int32 id = 2;
  string login = 3;
  string name = 4;
}

message NewUser{
  string login = 1;
  string name = 2;
  string password1 = 3;
  string password2 = 4;
}
